pynamodb/connection/base.py

```diff
@@ -923,6 +925,109 @@ def get_item(self,
+                          table_name,
+                          attributes_to_get=None,
+                          page_size=None,
+                          limit=None,
+                          conditional_operator=None,
+                          scan_filter=None,
+                          exclusive_start_key=None,
+                          segment=None,
+                          total_segments=None,
+                          time_out_seconds=None,
+                          read_capacity_to_consume_per_second=10,
+                          max_sleep_between_retry=10,
+                          max_consecutive_exceptions=10):
+        """
+        Performs a rate limited scan of the table.
+        """
+        read_capacity_to_consume_per_ms = float(read_capacity_to_consume_per_second) / 1000
+        total_consumed_read_capacity = 0.0
+        last_evaluated_key = exclusive_start_key
+        rate_available = True
+        latest_scan_consumed_capacity = 0
+        consecutive_provision_throughput_exceeded_ex = 0
+        start_time = time.time()
+
+        if page_size is None:
+            if limit and read_capacity_to_consume_per_second > limit:
+                page_size = limit
+            else:
+                page_size = read_capacity_to_consume_per_second
+
+        while True:
+            if rate_available:
+                try:
+                    data = self.scan(
+                        table_name,
+                        attributes_to_get=attributes_to_get,
+                        exclusive_start_key=last_evaluated_key,
+                        limit=page_size,
+                        conditional_operator=conditional_operator,
+                        return_consumed_capacity=TOTAL,
+                        scan_filter=scan_filter,
+                        segment=segment,
+                        total_segments=total_segments
+                    )
+                    for item in data.get(ITEMS):
+                        yield item
+
+                        if limit is not None:
+                            limit -= 1
+                            if not limit:
+                                return
+
+                    latest_scan_consumed_capacity = data.get(CONSUMED_CAPACITY)
+                    last_evaluated_key = data.get(LAST_EVALUATED_KEY, None)
+                    consecutive_provision_throughput_exceeded_ex = 0
+                except ScanError as e:
+                    # Only retry if provision throughput is exceeded.
+                    if isinstance(e.cause, ClientError):
+                        code = e.cause.response['Error'].get('Code')
+                        if code == "ProvisionedThroughputExceededException":
+                            consecutive_provision_throughput_exceeded_ex += 1
+                            if consecutive_provision_throughput_exceeded_ex > max_consecutive_exceptions:
+                                raise # Max threshold reached
```

Comment:
pep8 says two space before inline comment

i would put them on line above the raise though